Assets/Player/CameraControls.gd

```diff
@@ -0,0 +1,114 @@
+extends Node
+
+const ZOOM_IN_LIMIT = 5
+const ZOOM_OUT_LIMIT = 35
+const ZOOM_VALUE = 5
+const RAY_LENGTH = 1000
+
+const MOVE_SPEED = 10
+const MOVE_FASTER_MULT = 3
+
+export(NodePath) var origin_path : NodePath
+export(NodePath) var camera_path : NodePath
+export(NodePath) var rotation_y_path : NodePath
+
+onready var _origin : Spatial = get_node(origin_path)
+onready var _camera : Camera = get_node(camera_path)
+onready var _rotation_y : Spatial = get_node(rotation_y_path)
+onready var _viewport := get_viewport()
+onready var _viewport_size := _viewport.size
+onready var _viewport_aspect := _viewport_size.aspect()
+
+var _basis : Basis
+var _move_drag_start : Vector2
+var enabled : bool = true setget set_enabled, get_enabled
+
+func _ready():
+	if _viewport.connect("size_changed", self, "_on_viewport_size_changed") != OK:
+		push_error("Failed To Connect Viewport")
+	_basis = _get_basis()
+
+func _process(delta):
+	_move(delta)
+	_move_drag()
+
+
+func _input(event):
```

Comment:
While the GDScript style guide does state to use 2 lines between functions, Unknown Horizons uses 1 line.